src/v/cluster/metadata_dissemination_service.cc

```diff
@@ -373,7 +373,7 @@ ss::future<> metadata_dissemination_service::update_leaders_with_health_report(
   cluster_health_report report) {
     for (auto& node_report : report.node_reports) {
         co_await _leaders.invoke_on_all(
-          [node_report](partition_leaders_table& leaders) {
+          [&node_report](partition_leaders_table& leaders) {
```

Comment:
just discovered we can do `[&node_report = std::as_const(node_report)]`

Reply:
Oh yeah, this is a really good idea. I was thinking mostly about lifetime issues, but mutation is just as dangerous.

Making the `as_const` change.

Reply:
Or maybe `const auto&` is better? Gahh.

Reply:
I guess from a style point of view we could all benefit from defaulting to const as far out in scope as possible, here and in nested loops, etc... I was intrigued by as_const, but that's not an endorsement.

Reply:
Yes, I think `const auto&` here would be "most const" == "more better" so I'll go with that.

Reply:
Updated to a const auto reference in the iteration.